### PR TITLE
feat(tracker): add currency context, add transactionTotal to order tr…

### PR DESCRIPTION
**Pull request by dkonieczek**
…ansaction payload

### CODE REVIEW

**Review by chrisFrazier77 — DISMISSED** [review]

File: packages/snap-tracker/src/Tracker.ts [hunk 1]
```diff
@@ -89,6 +89,12 @@ export class Tracker {
 			},
 		};
 
+		if (this.globals.currency) {
```
Comment (korgon) [inline packages/snap-tracker/src/Tracker.ts:92]: `this.globals.currency` should be of type:
```
{
  code: string;
}
```

File: packages/snap-tracker/src/Tracker.ts [hunk 2]
```diff
@@ -546,6 +571,15 @@ export class Tracker {
 		}
 	};
 
+	setCurrency = (currency: string): void => {
```
Comment (korgon) [inline packages/snap-tracker/src/Tracker.ts:574]: This function should take an object as the param:
```
{
  code: string;
}
```

File: packages/snap-tracker/src/Tracker.ts [hunk 3]
```diff
@@ -418,7 +424,7 @@ export class Tracker {
 			},
 		},
 		cart: {
-			view: (data: CartViewEvent, siteId?: string): BeaconEvent | undefined => {
+			view: (data: CartViewEvent, siteId?: string, currency?: string): BeaconEvent | undefined => {
```
Comment (korgon) [inline packages/snap-tracker/src/Tracker.ts:427]: I think we can remove the currency param here and instead assume that the currency has already been set on the context.

File: packages/snap-tracker/src/Tracker.ts [hunk 4]
```diff
@@ -475,10 +490,10 @@ export class Tracker {
 			},
 		},
 		order: {
-			transaction: (data: OrderTransactionData, siteId?: string): BeaconEvent | undefined => {
+			transaction: (data: OrderTransactionData, siteId?: string, currency?: string): BeaconEvent | undefined => {
```
Comment (korgon) [inline packages/snap-tracker/src/Tracker.ts:493]: I think we can remove the currency param here and instead assume that the currency has already been set on the context.

**Review by korgon — CHANGES_REQUESTED** [review]
A few changes requested in review, but additionally, there should be a way to set the currency via the [script context variables](https://searchspring.github.io/snap/#/integration-context) (similar to `shopperId`).

Perhaps something like this:
```
<script src="https://snapui.searchspring.io/[your_site_id]/bundle.js">
    currency = { code: "USD" };
</script>
```

**Review by korgon — APPROVED** [review]
Looks good.

**Pull request merged**